### PR TITLE
Tiny fixes on lightning

### PR DESCRIPTION
**Pull request by zhentaocc**
- Carefully handle loss, metrics in LightningModule
- add `pred_step`
- Add argument `scheduler` to configure scheduler
- Add argument `scheduler` to compile scheduler through `trainer.compile`
- Rename metrics with indexes as suffix to avoid different metrics with same class

### CODE REVIEW

File: python/nano/src/bigdl/nano/pytorch/lightning.py [hunk 1]
```diff
@@ -56,30 +64,27 @@ def training_step(self, batch, batch_idx):
...
-        else:
-            acc = None
-        return loss, acc
+
+    def predict_step(self, batch, batch_idx):
+        return self(*batch)
 
     def configure_optimizers(self):
-        return self.optimizer
+        return [self.optimizer], [self.scheduler]
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/lightning.py:87]: is it ok to pass a scheduler that is `None`

**Comment by yangw1234** [comment]
LGTM, 
except that we need to pay attention whether a None works in `configure_optimizer`

**Comment by zhentaocc** [comment]
http://10.112.231.51:18888/job/ZOO-PR-NanoTests/333/

File: python/nano/src/bigdl/nano/pytorch/lightning.py (hunk 1 above)
Comment (zhentaocc) [reply python/nano/src/bigdl/nano/pytorch/lightning.py]: resolved

**Comment by yangw1234** [comment]
LGTM

**Pull request merged**